### ISSUE TITLE
One epoch hangs >60s for conv+svm on ordinal arrays_npz.npz on amd0

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#371]
Earliest public failure is recipe.train().run remaining active after 60 seconds on backend amd0. Inspected implementation cause is recipe.train().run -> estimator fitting for svm plus conv forward/backward with batch normalization. Measured 60s wall time exceeds defensible range for this workload. The configured model is
recipe.model()
.conv(8, 1).sigmoid().norm(batch).iq(2).m
.svm().norm(batch).qi(5).k.s
.loss(ce)
with recipe.data("data/ordinal/arrays_npz.npz").target("target") and recipe.train().optimizer(adamw).lr(0.0001).seed(115173441087118).epochs(1).fp(64) on amd0. fit_svm iterates 128 times over rows*features with regularized variance and affine predictor, and batch norm stats passes scan rows*channels*length, none of which justify 60s.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `67e54cd72bd639e1`
- Classifier provider: `opencode`
- Classifier model: `muse-spark-1.2-contributor-free`
- Classifier effort: `default`

### Classifier decision

```json
{
  "verdict": "new",
  "issue": 0,
  "title": "One epoch hangs >60s for conv+svm on ordinal arrays_npz.npz on amd0",
  "body": "Earliest public failure is recipe.train().run remaining active after 60 seconds on backend amd0. Inspected implementation cause is recipe.train().run -> estimator fitting for svm plus conv forward/backward with batch normalization. Measured 60s wall time exceeds defensible range for this workload. The configured model is\nrecipe.model()\n.conv(8, 1).sigmoid().norm(batch).iq(2).m\n.svm().norm(batch).qi(5).k.s\n.loss(ce)\nwith recipe.data(\"data/ordinal/arrays_npz.npz\").target(\"target\") and recipe.train().optimizer(adamw).lr(0.0001).seed(115173441087118).epochs(1).fp(64) on amd0. fit_svm iterates 128 times over rows*features with regularized variance and affine predictor, and batch norm stats passes scan rows*channels*length, none of which justify 60s.",
  "rationale": "Performance packet kind=performance shows stable 60s hang for one epoch on amd0. Workload is conv(8,1) with sigmoid batch norm iq(2).m followed by svm qi(5).k.s on data/ordinal/arrays_npz.npz. SVM cost is svm_iterations=128 * rows * features with per row prediction bias + weights dot (value-mean)*scale and gradient accumulation, plus two passes for mean/variance. Conv cost is single 8 filter kernel 1 contraction per sample. Even assuming upper bound 10k rows and 64 features, arithmetic is under 1e9 FLOPs and memory traffic tens of MB, expected 0.05 to 0.5s on CPU and less on amd0 GPU via native kernels. Throughput measured is <1/60 epoch per second vs expected >2 epochs per second. No existing issue read describes this earliest cause through the same SVM and batch norm implementation path, so new is required."
}
```

## Deterministic failure packet

kind=performance
id=67e54cd72bd639e1
base=commit=e7f8a6cb1b2f3c880309eefc6b7ebac7f717ea99 tracked_tree=modified
cursor=seed:0 cursor:16442 next:16443 composition:115173441087118
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:amd0
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-issue-machine/target/debug/recipe-trials/recipe-trial-455-16442/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/ordinal/arrays_npz.npz")
		.target("target");
	let model = recipe.model()
		.conv(8, 1).sigmoid().norm(batch).iq(2).m
		.svm().norm(batch).qi(5).k.s
		.loss(ce);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(115173441087118)
		.epochs(1)
		.log(all)
		.stop(0.8)
		.fp(64)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=amd0

**Comment by nm-z** [comment]
Closing as resolved based on zero-change verification PR #462.

That PR changes 0 files and verifies the existing `minimal` behavior, so there is no code to merge. The original verification is copied below.

<details>
<summary>Verification from PR #462: Verify current AMD convolution SVM path</summary>

Closes #371

The exact current public AMD reproduction completes within the reported 60-second boundary. No production change is supported by the current behavior.

Exact command:

```sh
/usr/bin/time -v env RECIPE_DEBUG=1 ./target/release/recipe --device amd0 /home/nate/.cache/recipe/issues/371/reproduction.rs
```

The NPZ contains 250,000 training rows, not an upper bound of 10,000. It has one signed 16-bit numeric input and one scalar target. The current path performs both SVM fits, including about 1.024 billion feature terms across their prediction and gradient loops.

Cold-cache result:

- Wall time: 5.75 seconds
- Epoch time: 1.6848 seconds
- BCE: 0.6931
- Maximum resident memory: 303,424 KB
- Exit status: 0

The cold root log records three real AMD native cache misses. Each completes AMD Clang compilation, native loading, and its required forward or epoch path. The convolution, SVM predictor, quantized storage, batch normalization, and FP64 epoch all finish normally.

Warm-cache result:

- Wall time: 3.91 seconds
- Epoch time: 1.6835 seconds
- BCE: 0.6931
- Maximum resident memory: 303,340 KB
- Exit status: 0

The cold and warm bundles are both 5,103,277 bytes and have the identical SHA-256 digest `3d7769a63481c46db5712029daa584430a493c55783055e0a0bdcbe783c9a879`.

The issue's only prior comment ran a probe-only installed binary with `recipe test.rs`; that command was rejected before loading the NPZ or reaching AMD execution.

Validation also passed `cargo fmt --check`, `git diff --check`, `cargo check`, and `cargo build --release --bin recipe`. This verification commit intentionally changes no tracked files.


</details>

Accepted verdict: PASS

Reason: The verification above reports that current `minimal` completes this issue's reported path successfully.
